Add pipelinesascode.tekton.dev/max-keep-runs annotation

chmouel opened the pull request:
There can be a lot of PipelineRuns into the user namespace. If the
PipelineRun has the an annotation :

```yaml
pipelinesascode.tekton.dev/max-keep-runs: "maxNumber"
```

after each Successfull Run Pipelinerun as Code will cleanup the old
PipelineRun attached to the same owner/repository and event type
(i.e: pull_request or push) and target branch (i.e: main) .

It will skip the Running PipelineRuns but will not skip the PipelineRuns
with `Unknown` status.

Signed-off-by: Chmouel Boudjnah <chmouel@redhat.com>

chmouel committed:
Disable gocyclo

vdemeester reviewed (commented):
This means it will keep all the pipelinerun that failed until one succeed right ?

chmouel commented:
@vdemeester yep, it only do the cleanup after a successful run. Rephrased a bit the documentation.

chmouel committed:
Add gofumpt target

I like it better and it is enforced in golangci

chmouel committed:
Add pipelinesascode.tekton.dev/max-keep-runs annotation

There can be a lot of PipelineRuns into the user namespace. If the
PipelineRun has the an annotation :

```yaml
pipelinesascode.tekton.dev/max-keep-runs: "maxNumber"
```

after each Successfull Run Pipelinerun as Code will cleanup the old
PipelineRun attached to the same owner/repository and event type
(i.e: pull_request or push) and target branch (i.e: main) .

It will skip the Running PipelineRuns but will not skip the PipelineRuns
with `Unknown` status.

Signed-off-by: Chmouel Boudjnah <chmouel@redhat.com>
